src/ui/src/writerApi.ts

```diff
@@ -391,6 +391,133 @@ export class WriterApi {
 		}
 		return res.text();
 	}
+
+	// Shared blueprints
+
+	#getAgentStorageBaseUrl(): string {
+		return (
+			import.meta.env.VITE_AGENT_STORAGE_URL ||
+			this.#baseUrl
+		);
+	}
+
+	async publishSharedBlueprint(
+		orgId: number,
+		data: {
+			title: string;
+			description: string;
+			components: unknown;
+			metadata: {
+				name?: string;
+				stateInputs?: string[];
+				stateOutputs?: string[];
+				vaultKeys?: string[];
+				dependencies?: unknown[];
+				author?: string;
+			};
+			existingSnippetId?: string | null;
+		},
+	): Promise<{ snippet_id: string; version: number }> {
+		const baseUrl = this.#getAgentStorageBaseUrl();
+		const url = new URL(
+			`/api/agent-storage/v1/organization/${orgId}/shared-blueprints`,
+			baseUrl,
+		);
+
+		const res = await fetch(url, {
+			...this.#requestInitBase,
+			method: "POST",
+			body: JSON.stringify({
+				title: data.title,
+				description: data.description,
+				components: data.components,
+				metadata: data.metadata,
+				existing_snippet_id: data.existingSnippetId || null,
+			}),
+		});
+		if (!res.ok) throw Error(await res.text());
+
+		return res.json();
+	}
+
+	async listSharedBlueprints(
+		orgId: number,
+		search?: string,
+	): Promise<
+		Array<{
+			id: string;
+			title: string;
+			description: string;
+			category: string;
+			version_number: number;
+			createdBy: number;
+		}>
+	> {
+		const baseUrl = this.#getAgentStorageBaseUrl();
+		const url = new URL(
+			`/api/agent-storage/v1/organization/${orgId}/shared-blueprints`,
+			baseUrl,
+		);
+
+		if (search) {
+			url.searchParams.append("search", search);
+		}
+
+		const res = await fetch(url, this.#requestInitBase);
+		if (!res.ok) throw Error(await res.text());
+
+		return res.json();
+	}
+
+	async getSharedBlueprint(
+		orgId: number,
+		blueprintId: string,
+	): Promise<{
+		id: string;
+		title: string;
+		visibility: string;
+		orgId: string;
+		createdBy: number;
+		version: {
+			version: number;
+			description: string | null;
+			components: unknown;
+			metadata: unknown;
+			createdAt: string;
+		};
+		createdAt: string;
+		updatedAt: string;
+	}> {
+		const baseUrl = this.#getAgentStorageBaseUrl();
+		const url = new URL(
+			`/api/agent-storage/v1/organization/${orgId}/shared-blueprints/${blueprintId}`,
+			baseUrl,
+		);
+
+		const res = await fetch(url, this.#requestInitBase);
+		if (!res.ok) throw Error(await res.text());
+
+		return res.json();
+	}
+
+	async deleteSharedBlueprint(
+		orgId: number,
+		blueprintId: string,
+	): Promise<{ success: boolean }> {
+		const baseUrl = this.#getAgentStorageBaseUrl();
+		const url = new URL(
+			`/api/agent-storage/v1/organization/${orgId}/shared-blueprints/${blueprintId}`,
+			baseUrl,
+		);
+
+		const res = await fetch(url, {
+			...this.#requestInitBase,
+			method: "DELETE",
+		});
+		if (!res.ok) throw Error(await res.text());
+
+		return res.json();
```

Comment:
Usually, DELETE returns an empty response `201`
```suggestion
	): Promise<void> {
		const baseUrl = this.#getAgentStorageBaseUrl();
		const url = new URL(
			`/api/agent-storage/v1/organization/${orgId}/shared-blueprints/${blueprintId}`,
			baseUrl,
		);

		const res = await fetch(url, {
			...this.#requestInitBase,
			method: "DELETE",
		});
		if (!res.ok) throw Error(await res.text());
```